docs/API.md

```diff
@@ -1,5 +1,49 @@
 # API
 
+## Table of contents
+
+* [Purpose](#purpose)
+* [Protocol](#protocol)
+  * [Deviations from JSON-RPC 2.0](#deviations-from-json-rpc-20)
+  * [Notes on the JSON-RPC 2.0 implementation](#notes-on-the-json-rpc-20-implementation)
+* [Request handling](#Request-handling)
+* [Error reporting](#Error-reporting)
+* [Privilege levels](#Privilege-levels)
+* [Data types](#Data-types)
+  * [API key](#API-key)
+  * [Batch id](#Batch-id)
+  * [Client id](#Client-id)
+  * [Client version](#Client-version)
+  * [Domain name](#Domain-name)
+  * [DS info](#DS-info)
+  * [IP address](#IP-address)
+  * [Language tag](#Language-tag)
+  * [Name server](#Name-server)
+  * [Non-negative integer](#Non-negative-integer)
+  * [Priority](#Priority)
+  * [Profile name](#Profile-name)
+  * [Progress percentage](#Progress-percentage)
+  * [Queue](#Queue)
+  * [Severity level](#Severity-level)
+  * [Test id](#Test-id)
+  * [Test result](#Test-result)
+  * [Timestamp](#Timestamp)
+  * [Username](#Username)
+* [API method: version_info](#API-method-version_info)
+* [API method: profile_names](#API-method-profile_names)
+* [API method: get_language_tags](#API-method-get_language_tags)
+* [API method: get_host_by_name](#API-method-get_host_by_name)
+* [API method: get_data_from_parent_zone](#API-method-get_data_from_parent_zone)
+* [API method: start_domain_test](#API-method-start_domain_test)
+* [API method: test_progress](#API-method-test_progress)
+* [API method: get_test_results](#API-method-get_test_results)
+* [API method: get_test_history](#API-method-get_test_history)
+* [API method: add_api_user](#API-method-add_api_user)
+* [API method: add_batch_job](#API-method-add_batch_job)
+* [API method: get_batch_job_result](#API-method-get_batch_job_result)
+* [API method: get_test_params](#API-method-get_test_params)
+
```

Comment:
https://github.com/zonemaster/zonemaster-backend/commit/e903cab0c143ff0e1c742c864be0c59994cb2dfa adds a new 2nd level heading "Undelegated and delegated". Should this be included in the toc? Or should the level of this new heading be 3 (###)?

Reply:
It should be level 3, and included. I will rebase this and undate. Good catch!

Reply:
This is then the only level 3 heading in the toc. Is this ok if we do not include all other level 3 headings? (For me this is fine, but just asking if we want to be consistent)

Reply:
All level 3 headings are included. There are several level 3 headings at the start of the document. All other headings under the methods are level 4 headings (which is a little strange since there are no level 3 headings).

Or would it be better to make it a level 4 heading like the other headings under the methods?

Reply:
Oops, you're right. I thought the `params`, `results`, ... were also level 3 headings. I think this is fine as it is now.